publication chapter

danielinteractive opened the pull request:
closes #9

Commit:
start publication chapter

Commit:
versions

Commit:
finish lecture content for publication

Commit:
first complete version of publication chapter

danielinteractive commented:
@fpahlke would be cool if you could chime in, maybe there is still something important missing - thank you :-)

fpahlke commented on slides/06_publication.qmd:
```diff
@@ -1,8 +1,189 @@
+
+## Tags
+
+- Tags are a feature of Git, i.e. not specific to GitHub
```
```suggestion
- Tags are a feature of Git, i.e., not specific to GitHub
```

fpahlke commented on slides/06_publication.qmd:
```diff
@@ -1,8 +1,189 @@
+- Reducing run time of tests, checks, examples, vignettes is important
+- Need to provide cross-platform portable code: 
+  CRAN runs checks on Windows, Mac, several Linux OS
+
```
```suggestion

## CRAN (cont'd)

A CRAN submission can be punishing, painful, and nerve-racking:

The first release of the [rpact](https://cran.r-project.org/package=rpact)
package took 5 weeks and 6 submission attempts; painful experiences:

-   Some Linux machines may generate different random numbers than expected
    (despite setting a seed)
-   Not only errors and warnings lead to the rejection of a submitted package,
    but also hints
-   Your local Windows test system may be much faster than the CRAN system,
    e.g., 5 times
-   Don't use "R Package for" in your package title
-   The description of the package must be provided with a doi reference

## CRAN (cont'd)

Example message informing about the rejection of the last
[rpact](https://cran.r-project.org/package=rpact) submission:

Dear maintainer,

package rpact_3.3.2.tar.gz does not pass the incoming checks automatically,
please see the following pre-tests: \
Windows:
[00check.log](https://cran.r-project.org/web/checks/check_results_rpact.html)
**Status: OK**, Debian:
[00check.log](https://cran.r-project.org/web/checks/check_results_rpact.html)
**Status: OK**

Please fix all problems and resubmit a fixed version via the webform.

Best regards, \
CRAN teams' auto-check service

r-devel-windows-x86_64 Check: Result: NA, Maintainer: 'Friedrich Pahlke'\
r-devel-windows-x86_64 Check: Overall checktime, Result: NOTE, Overall checktime
12 min \> 10 min\
r-devel-linux-x86_64-debian-gcc Check: Result: Note_to_CRAN_maintainers
Maintainer: 'Friedrich Pahlke'

```

fpahlke reviewed (approved):
Looks great 👍 

I only added two slides "CRAN (cont'd)" from my rpact submission experience...

danielinteractive committed:
Apply suggestions from code review

Co-authored-by: Friedrich Pahlke <44603414+fpahlke@users.noreply.github.com>

Commit:
final touches

danielinteractive commented:
@kkmann this would also be more or less ready for merge. maybe you also want to have a quick look first?

kkmann committed:
merged with main

Merge branch 'main' into 9_publication

# Conflicts:
#	docs/search.json
#	docs/sitemap.xml
